File: JavaPrototypes/ExpenseCalculator/tests/com/example/expenseCalculatorTests/EmployeeFactoryTests.java
package com.example.expenseCalculatorTests;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import com.example.expenseCalculator.Employee;
import com.example.expenseCalculator.EmployeeFactory;
import com.example.expenseCalculator.EmployeeType;

public class EmployeeFactoryTests {

	@Before
	public void setUp() throws Exception {
	}

	@After
	public void tearDown() throws Exception {
	}
	
	private static ExpenseCalculatorTestHelper testHelper = new ExpenseCalculatorTestHelper();
	private Pair<String, Double> _qatester = testHelper.findPairByString("QA Tester");
	private Pair<String, Double> _developer = testHelper.findPairByString("Developer");
	private Pair<String, Double> _manager = testHelper.findPairByString("Manager");

	@Test
	public void createQATester_Expense_Is_500() {
		Employee employee = EmployeeFactory.create(EmployeeType.Employee, _qatester.x, _qatester.y);
		
		double actual = employee.getExpenseAllocation();
		double expected = (double)_qatester.y;
		
		Assert.assertEquals(expected, actual, 0);
	}
	
	@Test
	public void createDeveloper_Expense_Is_1000() {
		Employee employee = EmployeeFactory.create(EmployeeType.Employee, _developer.x, _developer.y);
		
		double actual = employee.getExpenseAllocation();
		double expected = (double)_developer.y;
		
		Assert.assertEquals(expected, actual, 0);
	}
	
	@Test
	public void createManager_Expense_Is_300() {
		Employee employee = EmployeeFactory.create(EmployeeType.Employee, _manager.x, _manager.y);
		
		double actual = employee.getExpenseAllocation();
		double expected = (double)_manager.y;
		
		Assert.assertEquals(expected, actual, 0);
	}

}

File: DotNetPrototypes/ExpenseCalculatorTests/EmployeeFactoryTests.cs
using System;
using ExpenseCalculator;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace ExpenseCalculatorTests
{
    [TestClass]
    public class EmployeeFactoryTests
    {
        private static ExpenseCalculatorTestHelper testHelper = new ExpenseCalculatorTestHelper();
        private Tuple<string, decimal> _qatester = testHelper.Employees.Find(delegate(Tuple<string, decimal> t) { return t.Item1 == "QA Tester"; });
        private Tuple<string, decimal> _developer = testHelper.Employees.Find(delegate(Tuple<string, decimal> t) { return t.Item1 == "Developer"; });
        private Tuple<string, decimal> _manager = testHelper.Employees.Find(delegate(Tuple<string, decimal> t) { return t.Item1 == "Manager"; });

        [TestMethod]
        public void CreateQATester_Expense_Is_500()
        {

            IEmployee employee = EmployeeFactory.Create(EmployeeType.Employee, _qatester.Item1, _qatester.Item2);
            Assert.AreEqual(employee.ExpenseAllocation, _qatester.Item2);
        }

        [TestMethod]
        public void CreateDeveloper_Expense_Is_1000()
        {
            IEmployee employee = EmployeeFactory.Create(EmployeeType.Employee, _developer.Item1, _developer.Item2);
            Assert.AreEqual(employee.ExpenseAllocation, _developer.Item2);
        }

        [TestMethod]
        public void CreateManager_Expense_Is_300()
        {
            IEmployee employee = EmployeeFactory.Create(EmployeeType.Manager, _manager.Item1, _manager.Item2);
            Assert.AreEqual(employee.ExpenseAllocation, _manager.Item2);
        }
    }
}
